index.js

```diff
@@ -33,37 +34,43 @@ function plugin(opts) {
   // Execute the plugin.
   return function (files, metalsmith, done) {
     // Prepare the options.
-    var pattern = opts.pattern || 'layouts/**'
+    var directory = opts.directory || 'layouts'
+    var pattern = opts.pattern
+    var includes = opts.includes || 'includes/**'
     var defaultLayout = opts.default
+    var ext = opts.ext || ''
 
     // Retrieve all layouts.
     var templates = {}
-    var filesKeys = Object.keys(files)
-    var layouts = minimatch.match(filesKeys, pattern, {matchBase: true})
+    var filesKeys = minimatch.match(Object.keys(files), pattern, {matchBase: true})
+    var layoutKeys = fs.readdirSync(path.join(metalsmith._directory, directory))
+    var layouts = minimatch.match(layoutKeys, "!" + includes, {matchBase: true})
 
     /**
      * Compile the given layout and store it in templates.
      */
     function compileLayout(layout, done) {
       // Find which JSTransformer to compile with.
-      var transform = path.extname(layout).substring(1)
+      var transform = path.extname(layout).substring(1) || ext.substring(1)
       transform = getTransformer(transform)
       if (transform) {
+        var filename = path.join(metalsmith._directory, directory, layout)
+        filename = path.extname(filename) ? filename : filename + ext;
         // Retrieve the options for the JSTransformer.
         var options = extend({}, files[layout], {
-          filename: path.join(metalsmith._directory, metalsmith._source, layout)
+          filename: filename
         })
 
         // Compile the content.
-        var content = files[layout].contents.toString()
+        var content = fs.readFileSync(filename).toString()
```

Comment:
Unsure about reading out of the source directory. While I understand templates uses this pattern, I'd like to keep the as much as possible within Metalsmith's `files` process. Could we provide an additional boolean option to allowing for this?